static/app/views/settings/organizationIntegrations/featureToggle.tsx

```diff
@@ -0,0 +1,119 @@
+import {useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Client} from 'sentry/api';
+import Switch from 'sentry/components/switchButton';
+import type {IntegrationWithConfig, Organization} from 'sentry/types';
+
+export interface ToggleableFeature {
+  disabledReason: string;
+  help: string;
+  initialState: boolean;
+  label: string;
+  name: string;
+}
+
+interface Props {
+  configurations: IntegrationWithConfig[];
+  features: ToggleableFeature[];
+  onUpdate: (updatedConfigs: IntegrationWithConfig[]) => void;
+  organization: Organization;
+}
+
+export default function FeatureToggleUpdater({
+  configurations,
+  organization,
+  features,
+  onUpdate,
+}: Props) {
+  const [featureToggles, setFeatureToggles] = useState<{[key: string]: boolean}>(
+    features.reduce((acc, feature) => {
+      acc[feature.name] = feature.initialState;
+      return acc;
+    }, {})
+  );
+
+  useEffect(() => {
+    const baseEndpoint = `/organizations/${organization.slug}/integrations/`;
+    const api = new Client();
+    const updatedConfigs: IntegrationWithConfig[] = [];
+    configurations.forEach(config => {
+      if (!config.configData) {
+        return;
+      }
+
+      const integrationConfigEndpoint = `${baseEndpoint}${config.id}/`;
+      const updatedConfigData = {
+        ...config.configData,
+        toggleableFlags: featureToggles,
+      };
+
+      try {
+        api.request(integrationConfigEndpoint, {
+          method: 'POST',
+          data: updatedConfigData,
+        });
+      } catch (error) {
+        // Nothing to do here
```

Comment:
What kind of error can happen here? If the toggle is not successful we should reflect that somehow in the UI. Throw a notice maybe?

Reply:
If the call fails, we'll have to hold onto all the errors and show that correctly. I think at this point, there is no error handling on this component, so we may have to introduce that. Maybe a banner per error, or just a singular error?